From eba4ff36d230249472f9d63994fba5af4e1dc153 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Mon, 1 Nov 2010 14:15:25 +0800
Subject: [PATCH] ENGR00133442 NAND : remove redundant clock operations

The nfc->begin() or nfc->end() will do the relative clock
opertions. Enable clock or disable clock here is redundant.
So just remove it.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c
index 3893af8..181cb47 100644
--- a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c
+++ b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c
@@ -374,7 +374,6 @@ static void mil_select_chip(struct mtd_info *mtd, int chip)
 	struct gpmi_nfc_data  *this  = nand->priv;
 	struct mil            *mil   = &this->mil;
 	struct nfc_hal        *nfc   =  this->nfc;
-	struct clk            *clock =  this->resources.clock;
 
 	DEBUG(MTD_DEBUG_LEVEL2, "[gpmi_nfc select_chip] chip: %d\n", chip);
 
@@ -382,13 +381,11 @@ static void mil_select_chip(struct mtd_info *mtd, int chip)
 
 	if ((mil->current_chip < 0) && (chip >= 0)) {
 		gpmi_nfc_start_event_trace("> mil_select_chip");
-		clk_enable(clock);
 		nfc->begin(this);
 		gpmi_nfc_add_event("< mil_select_chip", -1);
 	} else if ((mil->current_chip >= 0) && (chip < 0)) {
 		gpmi_nfc_add_event("> mil_select_chip", 1);
 		gpmi_nfc_add_event("> not disable clk", 1);
-		clk_disable(clock);
 		nfc->end(this);
 		gpmi_nfc_stop_event_trace("< mil_select_chip");
 	} else {
-- 
1.5.4.4

